docs/src/main/asciidoc/dev-mode-differences.adoc

```diff
@@ -0,0 +1,96 @@
+during development but should *NEVER* be used in production.
+
+[[architectural-differences]]
+== Architectural differences
+
+Feature sets aside, the Quarkus application that is run under dev-mode differs architecturally from the production application (i.e. the one that is run using `java -jar ...`).
+
+In dev-mode, Quarkus uses a ClassLoader hierarchy (explained in detail link:class-loading-reference.adoc[here]) that enables the live reload of user code
+without requiring a rebuild and restart of the application.
+
+In a production application, the aforementioned class loading infrastructure is entirely absent - there is a single, purpose built ClassLoader that loads (almost) all classes and dependencies.
+
+== Dev-mode features
+
+In keeping with the mantra of providing developer joy, Quarkus provides a host of features when dev-mode is enabled. The most important features are:
+
+=== Live reload
+
+This mightily important feature needs no introduction and has already been mentioned in the <<architectural-differences, architectural differences>> section.
+
+=== Dev UI
+
+Quarkus provides a very useful link:dev-ui.adoc[UI] accessible from the browser at `/q/dev`. This UI allows a developer to see the state of the application, but
+also provides access to various actions that can change that state (depending on the extensions that are present).
+Examples of such operations are:
+
+* Changing configuration values
+* Running Database migration scripts
+* Clearing of caches
+* Running scheduled operations
+* Building a container
+
+=== Error pages
+
+In an effort to make development errors very easy to diagnose, Quarkus provides various detailed error pages when running in dev-mode.
+
+=== Database import scripts
+
+The `quarkus-hibernate-orm` extension will run the `import.sql` script in `src/main/resources` when Quarkus is running in dev-mode. More details can be found link:hibernate-orm.adoc#dev-mode[here].
+
+=== Dev Services
+
+When testing or running in dev-mode Quarkus can even provide you with a zero config database out of the box, a feature we refer to as DevServices.
+More information can be found link:datasource.adoc#dev-services[here].
+
+=== Swagger UI
+
+The `quarkus-smallrye-openapi` extension will expose the Swagger UI when Quarkus is running in dev-mode. Additional information can be found link:openapi-swaggerui.adoc#dev-mode[here].
+
+=== GraphQL UI
+
+The `quarkus-smallrye-graphql` extension will expose the GraphiQL UI when Quarkus is running in dev-mode. More details can be found link:smallrye-graphql.adoc#ui[here].
+
+=== Health UI
+
+The `quarkus-smallrye-health` extension will expose the Health UI when Quarkus is running in dev-mode. link:smallrye-health.adoc#ui[This] section provides additional information.
+
+=== Mock mailer
+
+The `quarkus-mailer` extension will enable an in-memory mock mail server when Quarkus is running in dev-mode. See link:mailer.adoc#testing[this] for more details.
+
+== Performance implications
```

Comment:
```suggestion

=== gRPC Reflection Service

The gRPC Reflection Service is enabled in dev mode by default. That lets you use tools such as `grpcurl`. In production mode, the reflection service is disabled. You can enable it explicitly using `quarkus.grpc-server. enable-reflection-service=true`

== Performance implications
```

Reply:
Is the extra space in the property name intentional or a typo? I think it's a typo?